e2e-tests/playwright/e2e/plugins/github-actions/github-actions.spec.ts

```diff
@@ -0,0 +1,98 @@
+import { expect, test } from "@playwright/test";
+import { UIhelper } from "../../../utils/ui-helper";
+import { Common } from "../../../utils/common";
+
+// Group the tests that test the GitHub Actions plugin functionality
+test.describe("Test GitHub Actions plugin functionality", () => {
+  // Reference component to test (should have GitHub Actions configured)
+  const componentWithGitHubActions = "Backstage Showcase";
+  
+  let uiHelper: UIhelper;
+  let common: Common;
+
+  test.beforeEach(async ({ page }) => {
+    // Increase default timeout to account for slower responses
+    test.setTimeout(180000); // 3 minutes
+    
+    common = new Common(page);
+    
+    // Login as GitHub user instead of guest
+    await common.loginAsGithubUser();
+    console.log("Logged in as GitHub user");
+    
+    uiHelper = new UIhelper(page);
+  });
+
+  test("Verify backstage-community-plugin-github-actions in Extensions page", async ({ page }) => {
+    // Navigate to Extensions page through Administration
+    await uiHelper.openSidebarButton("Administration");
+    await uiHelper.clickLink("Extensions");
+    console.log("Navigated to Extensions page");
+    
+    // Click on the Installed tab
+    await uiHelper.clickTab("Installed");
+    console.log("Clicked on Installed tab");
+    
+    // Wait for the page to fully load
+    await page.waitForLoadState('networkidle', { timeout: 30000 });
+    await page.waitForLoadState('domcontentloaded', { timeout: 30000 });
+    
+    // Wait for the table to load - using more reliable selector
+    await page.waitForSelector('table, div[role="grid"], [data-testid="plugins-table"]', { timeout: 30000 });
+    await page.waitForTimeout(3000); // Add a delay to ensure the table is fully rendered
```

Comment:
Using page.waitForLoadState('networkidle') and page.waitForLoadState('domcontentloaded') is not recommended for modern web applications. These methods rely on global states that can be unreliable in dynamic applications with ongoing network requests (e.g., WebSockets or API polling). Instead, it’s better to wait for specific elements or components that indicate the page is fully loaded and interactive. For example, use 
`page.waitForSelector('[data-testid="plugins-table"]')` to ensure tests are more reliable and maintainable.

Reply:
fixed